Split intro training into two separate modules (#377)
moderneinc/moderne-docs: docs/hands-on-learning/introduction/workshop-overview.md

---
sidebar_label: Workshop overview
description: A high-level overview of what this workshop will entail.
---

> # Overview: Introduction to Moderne's OpenRewrite

Author: Removed "Moderne's"

[OpenRewrite](https://docs.openrewrite.org/) is a framework for writing and running code transformations and [recipes](https://docs.openrewrite.org/concepts-and-explanations/recipes) are the unit of work in OpenRewrite. This workshop is designed to help you understand how [OpenRewrite](https://docs.openrewrite.org/) works and gain hands-on experience running recipes using the Moderne CLI and the Moderne Platform.

> You will learn about fundamental concepts like the [Lossless Semantic Tree (LST)](https://docs.openrewrite.org/concepts-and-explanations/lossless-semantic-trees), [data tables](https://docs.moderne.io/user-documentation/moderne-platform/getting-started/data-tables/), [visualizations](https://docs.moderne.io/user-documentation/moderne-platform/getting-started/visualizations), and [DevCenter](https://docs.moderne.io/user-documentation/moderne-platform/getting-started/dev-center). This workshop is designed to be hands-on, so you can follow along with the examples in your own environment. The workshop consists of a series of modules that highlight key concepts, each with one or more exercises to help you practice what you've learned.

Reviewer: This sentence has a category mismatch. For example, a visualization isn't a fundamental concept like an LST is. Furthermore the tenses don't flow properly. I'd propose something like this instead: 

```suggestion
You will learn about fundamental concepts like the [Lossless Semantic Tree (LST)](https://docs.openrewrite.org/concepts-and-explanations/lossless-semantic-trees), as well as key platform features including [data tables](https://docs.moderne.io/user-documentation/moderne-platform/getting-started/data-tables/), [visualizations](https://docs.moderne.io/user-documentation/moderne-platform/getting-started/visualizations), and [DevCenters](https://docs.moderne.io/user-documentation/moderne-platform/getting-started/dev-center).
```

Be sure to also follow the links to the [OpenRewrite documentation](https://docs.openrewrite.org/) for more in-depth information. Feel free to skip around to the sections that interest you most, based on your needs and experience level.

If you get stuck or have questions, feel free to ask in the [OpenRewrite Slack](https://join.slack.com/t/rewriteoss/shared_invite/zt-nj42n3ea-b~62rIHzb3Vo0E1APKCXEA) or [Discord](https://discord.gg/xk3ZKrhWAb).

## What you'll learn

* Key concepts of OpenRewrite: LST, visitors, and recipes
* How to set up and run recipes using the Moderne CLI
* Using the Moderne recipe builder to compose and customize declarative YAML recipes
* Understanding search markers, data tables, and visualizations

> * Using DevCenter to track migrations and upgrades

Reviewer: The phrasing is inconsistent in this list. If we think about the title "What you'll learn" - you can think of a sentence like "You will learn key concepts of OpenRewrite" or "You will learn how to set up and run recipes" - both of which make sense.

However, consider "You will learn using the Moderne recipe builder" or "You will learn understanding search markers." Those things don't make sense.

Please update this list for consistency.

## Prerequisites

To get the most out of this workshop, you should at least be familiar with:

* Basic Java knowledge

> * Command line/terminal basics

Reviewer: I don't think we should use the phrase "at least" - it feels a bit too harsh. Also there are the same issues with the bullets here as I mentioned above. Here is a suggestion for how to fix:

```suggestion
To get the most out of this workshop, you should have:

* Basic Java knowledge
* A basic understanding of how to work with the command line/terminal
```
